pkgs/games/xonotic/default.nix

```diff
@@ -62,7 +62,7 @@ let
       sha256 = "0axxw04fyz6jlfqd0kp7hdrqa0li31sx1pbipf2j5qp9wvqicsay";
     };
 
-    buildInputs = [ unzip libjpeg zlib libvorbis curl ]
+    buildInputs = [ unzip libjpeg zlib gmp libvorbis libtheora curl libpng freetype ]
```

Comment:
Are freetype, libtheora and so on really needed as buildInput for the dedicated server as well?

Reply:
Good question. I suppose not. I was told in the chat yesterday about the nifty <code>-developer</code> option for the executables which tells us what libraries are being attempted to be loaded. Here's the output for the dedicated and sdl-variant (glx doesn't get built on my system):

xonotic-dedicated:
<code>
Trying to load library... "libd0_blind_id.so.0" - loaded.
Trying to load library... "libd0_rijndael.so.0" - loaded.
Trying to load library... "libcurl.so.4" - loaded.
Skeletal animation uses SSE code path
Trying to load library... "libode.so.3" "libode.so.2" "libode.so.1" "/home/fps/.nix-profile/bin/libode.so.3" "/home/fps/.nix-profile/bin/libode.so.2" "/home/fps/.nix-profile/bin/libode.so.1" - failed.
</code>

xonotic-sdl:
<code>
Trying to load library... "libd0_blind_id.so.0" - loaded.
Trying to load library... "libd0_rijndael.so.0" - loaded.
Trying to load library... "libcurl.so.4" - loaded.
Skeletal animation uses SSE code path
Trying to load library... "libode.so.3" "libode.so.2" "libode.so.1" "/home/fps/.nix-profile/bin/libode.so.3" "/home/fps/.nix-profile/bin/libode.so.2" "/home/fps/.nix-profile/bin/libode.so.1" - failed.
Initializing client
Couldn't load gfx/palette.lmp, falling back on internal palette
DPSOFTRAST available (SSE2 instructions detected)
Failed to init SDL joystick subsystem: 
Trying to load library... "libvorbis.so.0" - loaded.
Trying to load library... "libvorbisfile.so.3" - loaded.
Trying to load library... "libogg.so.0" - loaded.
Trying to load library... "libtheora.so.0" - loaded.
Trying to load library... "libvorbis.so.0" - loaded.
Trying to load library... "libvorbisenc.so.2" - loaded.
Trying to load library... "libavw.so.1" "libavw.so" "/home/fps/.nix-profile/bin/libavw.so.1" "/home/fps/.nix-profile/bin/libavw.so" - failed.
</code>

It seems I was still missing ODE and AVW which might be required for some features..